File: onyx-web-database/src/com/onyx/server/DatabaseUserAccount.java
package com.onyx.server;

import com.onyx.entity.SystemUser;
import io.undertow.security.idm.Account;

import java.security.Principal;
import java.util.HashSet;
import java.util.Set;

/**
 * Created by timothy.osborn on 4/23/15.
 *
 * Auth account object
 */
class DatabaseUserAccount implements Account
{
    private final SystemUser user;
    private Set<String> roles = null;

    DatabaseUserAccount(SystemUser user)
    {
        this.user = user;
    }

    @Override
    public final Principal getPrincipal()
    {
        return user;
    }

    @Override
    public final Set<String> getRoles()
    {
        if(user == null)
        {
            return null;
        }
        if(roles == null)
        {
            roles = new HashSet<>();
            if(user.getRole() != null)
            {
                roles.add(user.getRole().name());
            }
        }
        return roles;
    }

    final String getPassword()
    {
        return user.getPassword();
    }
}

File: onyx-web-database/src/main/kotlin/com/onyx/server/DatabaseUserAccount.kt
package com.onyx.server

import com.onyx.entity.SystemUser
import io.undertow.security.idm.Account

import java.security.Principal
import java.util.HashSet

/**
 * Created by timothy.osborn on 4/23/15.
 *
 * Auth account object
 */
internal class DatabaseUserAccount(private val user: SystemUser?) : Account {

    private var roles: MutableSet<String>? = null

    val password: String?
        get() = user!!.password

    override fun getPrincipal(): Principal? = user

    override fun getRoles(): Set<String>? {
        if (user == null) {
            return null
        }
        if (roles == null) {
            roles = HashSet()
            if (user.role != null) {
                roles!!.add(user.role!!.name)
            }
        }
        return roles
    }

}
